sklearn/base.py

```diff
@@ -187,6 +187,25 @@ class BaseEstimator(_HTMLDocumentationLinkMixin, _MetadataRequester):
     array([3, 3, 3])
     """
 
+    def freeze(self):
+        """Return a frozen version of this estimator.
+
+        This method wraps the estimator in a :class:`~sklearn.frozen.FrozenEstimator`
+        object, and returns the frozen object. This method does NOT freeze this object
+        itself.
+
+        The return frozen object will prevent changes to this estimator's instance
+        parameters; however, direct changes to this estimator are still possible.
+
+        Returns
+        -------
+        frozen : :class:`~sklearn.frozen.Frozen`
+            A frozen version of this estimator.
+        """
+        from sklearn.frozen import FrozenEstimator
```

Comment:
Is this necessary to avoid circular imports between `sklearn.frozen` and `sklearn.base`? If so, we should probably add a comment.

If not, why not move that to the list of top-level import statements? Importing `sklearn.frozen` should not be too costly, no? I don't expect any heavy dynamically linked compiled extensions to be loaded as a result of this extra import, right?